File: python/staking.py
from ftx.client import FtxClient


COINS = ['SRM', 'SRM_LOCKED', 'MSRM', 'MSRM_LOCKED', 'SOL', 'RAY']


def auto_staking(client: FtxClient) -> None:
    for coin in COINS:
        stake_coin(client, coin)


def stake_coin(client: FtxClient, coin: str) -> None:
    balance_coin = client.get_balance_coin(coin)
    if balance_coin is None:
        return
    available_balance = balance_coin['free']
    if available_balance > 0:
        client.stakes(coin=coin, size=available_balance)

File: go/staking.go
package main

import (
	"log"
	. "stakeSrm/ftx"
	"sync"
	"time"
)

var COINS = []string{"SRM", "SRM_LOCKED", "MSRM", "MSRM_LOCKED", "SOL", "RAY"}

func auto_staking(client *FtxClient, wg *sync.WaitGroup) {
	defer wg.Done()
	for {
		for _, coin := range SRM_COINS {
			stake_coin(client, coin)
			time.Sleep(time.Second * 60 * 60)
		}
	}
}

func stake_coin(client *FtxClient, coin string) {
	available_balance, err := client.GetAvalaibleBalanceCoin(coin)
	if err != nil {
		log.Printf("Error getting available SRM %s", err)
	}
	if available_balance > 0 {
		_, err := client.Stakes(coin, available_balance)
		if err != nil {
			log.Printf("Error staking SRM %s", err)
		}
	}
}
